src/Workspaces/Core/Portable/LanguageServices/SyntaxFactsService/AbstractSyntaxFactsService.cs

```diff
@@ -95,10 +99,144 @@ protected static void FreeAliasMap(Dictionary<string, string> aliasMap)
         }
 
         protected static Dictionary<string, string> AllocateAliasMap()
+            => s_aliasMapPool.Allocate();
+
+        public bool IsOnSingleLine(SyntaxNode node, bool fullSpan)
```

Comment:
Why isn't this just asking for the line numbers for the start and end of the node's span, and comparing?